src/test/util/setup_common.h

```diff
@@ -45,6 +45,23 @@ std::ostream& operator<<(typename std::enable_if<std::is_enum<T>::value, std::os
 {
     return stream << static_cast<typename std::underlying_type<T>::type>(e);
 }
+
+// Enable BOOST_CHECK_EQUAL for std::optional
```

Comment:
Pulled these to another PR: [`d1e3f0e` (#30618)](https://github.com/bitcoin/bitcoin/pull/30618/commits/d1e3f0e8a9df410317a120b81087fa4a0cddc480)

note1: nit: `// Enable BOOST_CHECK_EQUAL for enum class types` could be put inside the `std` now that we have multiple
note2: we might want to move the remaining ones in the cpp file here as well (inline-ing them all, of course)

Reply:
Was merged since (added you as co-author), you should be able to remove this now